src/Build/Graph/ProjectGraph.cs

```diff
@@ -22,21 +28,69 @@ internal sealed class ProjectGraph
         /// <exception cref="InvalidProjectFileException">If the evaluation of any project in the graph fails.</exception>
         public ProjectGraph(string entryProjectFile)
         {
-            var graphNode =
-                new ProjectGraphNode(ProjectCollection.GlobalProjectCollection.LoadProject(entryProjectFile));
-            _projectNodes.Add(graphNode);
+            AddNewNode(entryProjectFile);
```

Comment:
This doesn't recurse. Seems like it should call the other ctor overload, or the `LoadGraph` directly.